src/String/Extra.elm

```diff
@@ -202,7 +195,7 @@ softBreak width string =
 
 softBreakRegexp : Int -> Regex.Regex
 softBreakRegexp width =
-    regexFromString <| ".{1," ++ String.fromInt width ++ "}(\\s+|$)|\\S+?(\\s+|$)"
+    regexFromString <| ".{0," ++ String.fromInt (width - 1) ++ "}(\\s|$)|\\S+?(\\s|$)"
```

Comment:
I played with this a bit last night and came to largely the same conclusion, but I wasn't entirely sure what the output was supposed to look like, since I couldn't reproduce the output from the comment. Even with your changes, it doesn't match so I think the comment is wrong?

```
---- Elm 0.19.1 ----------------------------------------------------------------
Say :help for help and :exit to exit! More at <https://elm-lang.org/0.19.1/repl>
--------------------------------------------------------------------------------
> import String.Extra exposing (softBreak)
> softBreak 6 "The quick brown fox"
["The ","quick ","brown ","fox"]
    : List String
```

Reply:
Yeah the comment is wrong (yet another place elm-verify-examples could have saved the day). It's kind of hard to understand what the desired output here should be. Is it "break the length and add any extra letters until reaching end-of-word" or is it "break text such that it would mostly fit into a column of that width, unless impossible"?

Because if it's the second, than `["The ","quick ","brown ","fox"]` is probably superior output to `[ "The quick", " brown", " fox" ]`.

Reply:
I *think* the idea is that it will break the string into a list of words, where each item in the list is at most `width` length. So a width of `6` wouldn't match `The quick` since that's 9, so it takes `The ` as the first item; similarly `quick brown ` and `brown fox` wouldn't match since they're too long as well. 

However, the list items can be shorter that `width`. 

I don't think the regex needs two capture groups, though, which is where I left off in my experiments. I found the origins of the function in an [older repo](https://github.com/lorenzo/elm-string-addons) but no word on the origins of the function. I was hoping that I could find a similar one in another language so I could check the two.